runtime/posix/src/main/java/org/qbicc/runtime/posix/SysStat.java

```diff
@@ -73,12 +73,21 @@ public static final class struct_stat64_ptr extends ptr<struct_stat64> {}
     public static final mode_t S_IWOTH = constant();
     public static final mode_t S_IXOTH = constant();
 
-    @name(value = "stat$INODE64", when = Build.Target.IsMacOs.class)
+    @name(value = "stat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
+    public static native c_int stat_inode64(const_char_ptr pathName, struct_stat_ptr statBuf);
+    @name(value = "fstat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
+    public static native c_int fstat_inode64(c_int fd, struct_stat_ptr statBuf);
+    @name(value = "lstat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
+    public static native c_int lstat_inode64(const_char_ptr pathName, struct_stat_ptr statBuf);
+    @name(value = "fstatat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
+    public static native c_int fstatat_inode64(c_int dirFd, const_char_ptr pathName, struct_stat_ptr statBuf, c_int flags);
+
+    @name(value = "stat", when = { Build.Target.IsMacOs.class, Build.Target.IsAarch64.class })
     public static native c_int stat(const_char_ptr pathName, struct_stat_ptr statBuf);
-    @name(value = "fstat$INODE64", when = Build.Target.IsMacOs.class)
+    @name(value = "fstat", when = { Build.Target.IsMacOs.class, Build.Target.IsAarch64.class })
```

Comment:
```suggestion
    @name(value = "fstat$INODE64", when = { Build.Target.IsMacOs.class, Build.Target.IsAmd64.class })
```

Wouldn't this suffice to solve the problem?

Reply:
Well for darwin/aarch64 yeah... but amd64?

Reply:
All this means is, when the platform is amd64, use the external name `fstat$INODE64` for this method instead of `fstat`. We don't need actual declarations for both, we just need the method to point to the correct function.

Reply:
N.b. if no `@name` is given, then the name of the method is used.